graql/reasoner/query/ReasonerQueryImpl.java

```diff
@@ -245,8 +258,6 @@ public int hashCode() {
     @Override
     public void checkValid() {
         getAtoms().forEach(Atomic::checkValid);
-        //this creates the corresponding substitution and checks for id validity
-        ConceptMap substitution = getSubstitution();
```

Comment:
just a note, this removes id validation. If that's expected then cool.

Reply:
Aha, this is because `getSubstitution()` throws if ids are not found. No this was not intended.